### DISCUSSION TITLE
Convert Single TW.html File to a Wiki Folder and, Customising Tiddler File Naming

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#7452] [General]
The below command is used to create a wiki fiolder from a single html file

```
tiddlywiki --load path/to/single/file.html --savewikifolder path/to/output/folder
```

The question is can we use a config like the `$:/config/FileSystemPaths` or a command parameter to customize the tiddler file name? For example I like `$__plugins_kookma_kara_history.tid`  saved as `plugins/kookma/kara/history.tid`

**Comment by pmario** [comment]
tiddlers will always be saved in the `/tiddlers` directory. So it's not possible to create plugins with custom path configs. I would have security concerns if it would be possible for the server to leave the /tiddlers subdirectory. 

It should be possible to create a `tiddlers/plugins` directory and then manually move it to a different plugins directory, that would be visible for `tiddlywiki.info` plugins section in the .info file

**Reply by kookma** [reply]
> tiddlers will always be saved in the `/tiddlers` directory.

Fortunately, or unfortunately for me this is not the case. Example

- download empty.html from https://tiddlywiki.com/
- open it in the browser add few plugins (by drag and drop or installing from a library)
  - here for demonstration I install an official plugin (codemirror) and a third party one (favorites)
- from command window convert it to node.js using `tiddlywiki --load empty.html --savewikifolder myempty01`
- now watch the `myempty01` folder
```
- myempty01
- -- plugins
- -- -- favorites
- -- tiddlers
```

So I have the plugins folder alongside with tiddlers folder in my wiki folder.

NOTE i:
- the Codemirror is not extracted at all, but it is defined in tiddlywiki.info
- the favorites is extracted under `plugins/favorites` folder

NOTE ii: I have a working TiddlyWIki on Node.JS with all plugins (official and third party in the path)

**Comment by Jermolene** [comment]
Hi @kookma the `--savewikifolder` command does obey any configuration specified in `$:/config/FileSystemExtensions` and `$:/config/FileSystemPaths`, and so I think that that idea might be workable.

However, as @pmario suggests, it is possible that this might represent a security risk: I think it might be possible for an attacker to craft a TiddlyWiki HTML file that when saved via `--savewikifolder` could overwrite arbitrary files with content of the attackers choosing. If so, that would warrant an issue.

**Reply by kookma** [reply]
> Hi @kookma the `--savewikifolder` command does obey any configuration specified in `$:/config/FileSystemExtensions` and `$:/config/FileSystemPaths`, and so I think that that idea might be workable.

Much appreciated! This works as expected, so I can create a folder structure in my wikifolder.
